features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Create/Default/APICreateDefault.jsx

```diff
@@ -198,45 +201,112 @@ function APICreateDefault(props) {
     }
 
     /**
-     *
      *
      */
     function createAndPublish() {
-        setIsPublishing(true);
-        createAPI().then((api) => api
-            .publish()
-            .then((response) => {
-                const { workflowStatus } = response.body;
-                if (workflowStatus === APICreateDefault.WORKFLOW_STATUS.CREATED) {
-                    Alert.info(intl.formatMessage({
-                        id: 'Apis.Create.Default.APICreateDefault.success.publishStatus',
-                        defaultMessage: 'Lifecycle state change request has been sent',
-                    }));
-                } else {
-                    Alert.info(intl.formatMessage({
-                        id: 'Apis.Create.Default.APICreateDefault.success.otherStatus',
-                        defaultMessage: 'API updated successfully',
-                    }));
-                }
-                history.push(`/apis/${api.id}/overview`);
-            })
-            .catch((error) => {
-                if (error.response) {
-                    Alert.error(error.response.body.description);
-                    setPageError(error.response.body);
-                } else {
-                    const message = 'Something went wrong while publishing the API';
-                    Alert.error(intl.formatMessage({
-                        id: 'Apis.Create.Default.APICreateDefault.error.errorMessage',
-                        defaultMessage: message,
-                    }));
-                    setPageError(message);
-                }
-                console.error(error);
-            })
-            .finally(() => {
-                setIsPublishing(false);
-            }));
+        const restApi = new API();
+        setIsPublishButtonClicked(true);
+        createAPI().then((api) => {
+            setIsRevisioning(true);
+            const body = {
+                description: 'Initial Revision',
+            };
+            restApi.createRevision(api.id, body)
+                .then((api1) => {
```

Comment:
Shall we use a more descriptive name here like `newAPI` rather than just appending a number